### ISSUE TITLE
Add `.gitattributes` and `.git/config` in `bun init` to make `git diff` just work for bun.lockb files

### ISSUE THREAD

**Issue by Jarred-Sumner** [oven-sh/bun#8274]
Into `.gitattributes`:
```js
*.lockb binary diff=lockb
```

Into `.git/config`:

```
[diff "lockb"]
textconv = bun
binary = true
```

**Comment by wojtekmaj** [comment]
Would be fantastic to also talk to:
* GitHub
* Gitlab
* Azure DevOps
* npm, Yarn, pnpm

maintainers to update the defaults set as you create repo with them.

**Comment by paperclover** [comment]
is `.git/config` synced to the remote. if not, then writing to it will cause a very confusing situation where one developers git environment is different from another without obvious reason why (they wont realize .git/config is written to)

**Comment by aboqasem** [comment]
> is `.git/config` synced to the remote. if not, then writing to it will cause a very confusing situation where one developers git environment is different from another without obvious reason why (they wont realize .git/config is written to)

Yeah this is where I was unsure, since `.git/config` is untracked, one would lose the config if cloned the `init`ed project later.

For me, [I add a `.gitconfig` file to the repo](https://twitter.com/mbalzouabi/status/1748155946684362774).

Should we generate a `.gitconfig` and then include it in `.git/config`?

```gitconfig
[include]
path = ../.gitconfig
```

Or just add a `postinstall` script: `git config --local include.path ../.gitconfig`?

**Comment by MRDGH2821** [comment]
Would be great if VSCode extension also gets this upgrade.
The diff shows up like this:
![image](https://github.com/oven-sh/bun/assets/46621001/1a567cba-8da5-4eb0-8c6a-bd980976ceea)

(This section may not show any difference, but those do exist at the bottom of this but my point is that VSCode is not highlighting which lines have difference. Maybe I'm naive because I do not know the internal working of this matter 🤔)

**Comment by thecaralice** [comment]
It does not "just work" because bun emits a message about loaded environment variables and screws up the pager.
<img width="216" alt="image" src="https://github.com/user-attachments/assets/44a0ecf9-9270-4ea1-aeb5-a623995aa34c">

**Comment by Jarred-Sumner** [comment]
> It does not "just work" because bun emits a message about loaded environment variables and screws up the pager. <img alt="image" width="216" src="https://private-user-images.githubusercontent.com/43097806/364577029-44a0ecf9-9270-4ea1-aeb5-a623995aa34c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU0OTM2NTIsIm5iZiI6MTcyNTQ5MzM1MiwicGF0aCI6Ii80MzA5NzgwNi8zNjQ1NzcwMjktNDRhMGVjZjktOTI3MC00ZWExLWFlYjUtYTYyMzk5NWFhMzRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA0VDIzNDIzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1MzI3ODMzZDJkNzljYzg4MTAyOGM2ZWUxNGIxOTc3YmJkMDYwMmI1YWFkOGE3OTkzNzZiMTBmNGM3NjExZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.J-2qb1IeVFxQ-5Ir1Agj2VsdiI93zpIF66O4E6fpYP4">

that's a bug, it shouldn't be printing a message for that by default anywhere

can you file an issue?

**Comment by thecaralice** [comment]
@Jarred-Sumner reported in #13743

**Comment by RiskyMH** [comment]
With `bun.lock` this isn't needed as it doesn't generate `bun.lockb` by default anymore
